### DISCUSSION TITLE
notcurses 3.0.0 will be out in a few days

### DISCUSSION THREAD

**Discussion by dankamongmen** [dankamongmen/notcurses#2393] [General]
please, please, please, please go hammer on it in any way you can, and file bug reports. please.

**Comment by joseluis** [comment]
Maybe we should homogenize the names of certain parameters?
- `ystart`, `xstart` → `begy`, `begx` (or do they have different meaning)?
- `ylen`, `xlen` → `leny`, `lenx` (or the reverse)

sometimes the x,y is at the beginning of the word, sometimes at the end...

**Reply by dankamongmen** [reply]
variable names aren't really part of the ABI -- they're beta-equivalent from a lambda calculus perspective. but feel free to submit a PR!

**Reply by joseluis** [reply]
Ok but I don't wanna change your variables unilaterally, what version do you prefer?

**Reply by dankamongmen** [reply]
i honestly have no preference whatsoever. maybe leading [yx]?

**Reply by joseluis** [reply]
I count 418 occurrences of `leny` vs 286 of `ylen` in the whole codebase. Also I think I like traling [yx] more, so it reads as written (len (of) y, and also for compound words like `keepleny`) and so the struct fields for length all starts by `len` and are more readily recognized as a group when reading left to right... I'll try to submit a PR...

**Reply by dankamongmen** [reply]
ok, but if you're gonna do that, please update `USAGE.md` and the man pages to match. also, know that the header file variable names don't have to match the actual C definitions, since they're not part of the API.

**Reply by joseluis** [reply]
so is it OK if I update every occurrence in every file? or just the public API & docs?

**Reply by joseluis** [reply]
another issue: does `starty` has to go together with `stopy` and `begy` with `leny`? or should we unify?

**Reply by joseluis** [reply]
another issue: I would unify `section` and `region` references to `region` in the case of visual & plane regions, and leave sections just for menu sections.

**Reply by joseluis** [reply]
> so is it OK if I update every occurrence in every file? or just the public API & docs?

There's a lot of places in the inner API, and more variables (ymax, ymin, yoff, ycell, ypic...), so I'm not gonna touch that. just the public API, less than 20 files vs more than 50...

**Reply by dankamongmen** [reply]
i think stopy and leny are different things? like if you're at 10, 10 and stop at 11, 11, that's a len of 1,1.


i'm kinda nervous about this. i don't honestly see any benefit, and it's going to be a tremendous amount of churn right before the release. and these *aren't part of the api*. we can change them whenever.

**Reply by joseluis** [reply]
> i think stopy and leny are different things? like if you're at 10, 10 and stop at 11, 11, that's a len of 1,1.

yeah, totally, I meant more the begy starty, which I do think is the same.

> i'm kinda nervous about this. i don't honestly see any benefit, and it's going to be a tremendous amount of churn right before the release. and these _aren't part of the api_. we can change them whenever.

:) ok I totally understand.. I'm gonna pospone this for a less stressful time

**Reply by dankamongmen** [reply]
thanks that's one fewer thing to worry about


feel absolutely encouraged to hammer on some testing!!! =]

**Reply by joseluis** [reply]
> another issue: I would unify `section` and `region` references to `region` in the case of visual & plane regions, and leave sections just for menu sections.

this is the only thing I've done, since it's a beneficial little thing

**Comment by joseluis** [comment]
can `margin_[trbl]` fields from `notcurses_options` & `ncplane_options` be negative? if not maybe make them unsigned.

**Reply by dankamongmen** [reply]
done

**Reply by joseluis** [reply]
now only remains `ncplane_options` :smile:

**Reply by dankamongmen** [reply]
y/x there need be signed

**Reply by dankamongmen** [reply]
rows and cols i made unsigned last week or whenever

**Reply by joseluis** [reply]
> y/x there need be signed

I'll trust you on that one, but could you confirm whether negative margin values are actually valid in there?

**Reply by dankamongmen** [reply]
y/x need to be signed because they're relative to the parent plane. oh i see what you mean, good call.

**Reply by dankamongmen** [reply]
done

**Comment by dankamongmen** [comment]
@Mukundan314 just made my year, and is a hero
